sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -425,14 +425,18 @@ object IntervalUtils {
   }
 
   private object ParseState extends Enumeration {
+    type ParseState = Value
+
     val PREFIX,
-        BEGIN_VALUE,
-        PARSE_SIGN,
-        PARSE_UNIT_VALUE,
-        FRACTIONAL_PART,
-        BEGIN_UNIT_NAME,
-        UNIT_NAME_SUFFIX,
-        END_UNIT_NAME = Value
+        NEXT_VALUE_UNIT,
```

Comment:
how about `TRIM_BEFORE_SIGN` to be consistent?

Reply:
cool. good naming